src/coreclr/jit/codegenlinear.cpp

```diff
@@ -1617,6 +1617,14 @@ void CodeGen::genConsumeRegs(GenTree* tree)
             genConsumeRegs(tree->gtGetOp1());
             genConsumeRegs(tree->gtGetOp2());
         }
+        else if (tree->OperIsFieldList())
+        {
+            for (GenTreeFieldList::Use& use : tree->AsFieldList()->Uses())
+            {
+                GenTree* fieldNode = use.GetNode();
+                genConsumeReg(fieldNode);
```

Comment:
```suggestion
                genConsumeRegs(fieldNode);
```
This function is not supposed to assume whether the child is node is contained or not.

Reply:
done.